### PR TITLE
[libra framework] add doc comments for vector, option, lcs modules

### PR DESCRIPTION
**Pull request by sblackshear**
Tweaking the comments that were already there and making them into doc comments.

### CODE REVIEW

File: language/stdlib/modules/Vector.move [hunk 1]
```diff
@@ -1,41 +1,45 @@
...
 
+    /// Create an empty vector.
     native public fun empty<Element>(): vector<Element>;
 
-    // Return the length of the vector.
+    /// Return the length of the vector.
     native public fun length<Element>(v: &vector<Element>): u64;
 
-    // Acquire an immutable reference to the ith element of the vector.
+    /// Acquire an immutable reference to the `i`th element of the vector `v`.
```
Comment (tzakian) [inline language/stdlib/modules/Vector.move:14]: Nit maybe add the same "aborts if i is out of bounds" comment here as in `borrow_mut`

File: language/stdlib/modules/Vector.move [hunk 2]
```diff
@@ -85,8 +89,8 @@ module Vector {
     }
 
 
-    // Remove the `i`th element E of the vector, shifting all subsequent elements
-    // It is O(n) and preserves ordering
+    /// Remove the `i`th element of the vector `v`, shifting all subsequent elements.
+    /// This is O(n) and preserves ordering of elements in the vector.
```
Comment (tzakian) [inline language/stdlib/modules/Vector.move:93]: Nit: Add "aborts if i  is out of bounds"

File: language/stdlib/modules/Vector.move [hunk 3]
```diff
@@ -97,8 +101,8 @@ module Vector {
         pop_back(v)
     }
 
-    // and then popping it off
-    // It is O(1), but does not preserve ordering
+    /// Swap the `i`th element of the vector `v` with the last element and then pop the vector.
+    /// This is O(1), but does not preserve ordering of elements in the vector.
```
Comment (tzakian) [inline language/stdlib/modules/Vector.move:105]: Nit: add "aborts if i is out of bounds"

File: language/stdlib/modules/Vector.move [hunk 4]
```diff
@@ -72,8 +76,8 @@ module Vector {
         false
     }
 
-    // Return (true, i) if `e` is in the vector `v` at index `i`.
-    // Otherwise returns (false, 0).
+    /// Return `(true, i)` if `e` is in the vector `v` at index `i`.
+    /// Otherwise, returns `(false, 0)`.
     public fun index_of<Element>(v: &vector<Element>, e: &Element): (bool, u64) {
```
Comment (tzakian) [inline language/stdlib/modules/Vector.move:81]: A hold-over from the non-option days...I'll file an issue to change this to an option.

Comment (sblackshear) [reply language/stdlib/modules/Vector.move:81]: SGTM, although the fact that `Option` is implemented using `Vector` may make this tricky...

Comment (tzakian) [reply language/stdlib/modules/Vector.move:81]: oh....you're right...nevermind! I keep forgetting that our options aren't really options.

**Pull request closed without merging**